exchanges/okx/okx.go

```diff
@@ -0,0 +1,4496 @@
+	params := url.Values{}
+	instrumentType = strings.ToUpper(instrumentType)
+	if !(instrumentType == "CONTRACTS" ||
+		instrumentType == okxInstTypeSpot) {
+		return nil, errMissingRequiredArgInstType
+	} else if instrumentType == okxInstTypeFutures ||
+		instrumentType == okxInstTypeMargin ||
+		instrumentType == okxInstTypeSwap ||
+		instrumentType == okxInstTypeOption {
+		return nil, fmt.Errorf("instrument type %s is not allowed for this query", instrumentType)
+	}
+	params.Set("instType", strings.ToUpper(instrumentType))
+	interval := ok.GetIntervalEnum(period)
+	if interval != "" {
+		params.Set("period", interval)
+	}
+	if currency != "" {
+		params.Set("ccy", currency)
+	}
+	if !begin.IsZero() {
+		params.Set("begin", strconv.FormatInt(begin.UnixMilli(), 10))
+	}
+	if !end.IsZero() {
+		params.Set("end", strconv.FormatInt(end.UnixMilli(), 10))
+	}
+	var response [][3]string
+	err := ok.SendHTTPRequest(ctx, exchange.RestSpot, getTakerVolumeEPL, http.MethodGet, common.EncodeURLValues(tradingTakerVolume, params), nil, &response, false)
+	if err != nil {
+		return nil, err
+	}
+	takerVolumes := []TakerVolume{}
+	for x := range response {
+		timestamp, err := strconv.Atoi(response[x][0])
+		if err != nil {
+			return nil, err
+		}
+		sellVolume, err := strconv.ParseFloat(response[x][1], 64)
+		if err != nil {
+			return nil, err
+		}
+		buyVolume, err := strconv.ParseFloat(response[x][2], 64)
+		if err != nil {
+			return nil, err
+		}
+		takerVolume := TakerVolume{
+			Timestamp:  time.UnixMilli(int64(timestamp)),
+			SellVolume: sellVolume,
+			BuyVolume:  buyVolume,
+		}
+		takerVolumes = append(takerVolumes, takerVolume)
+	}
+	return takerVolumes, nil
+}
+
+// GetMarginLendingRatio retrieves the ratio of cumulative amount between currency margin quote currency and base currency.
+func (ok *Okx) GetMarginLendingRatio(ctx context.Context, currency string, begin, end time.Time, period kline.Interval) ([]MarginLendRatioItem, error) {
+	params := url.Values{}
+	if currency != "" {
+		params.Set("ccy", currency)
+	}
+	if !begin.IsZero() {
+		params.Set("begin", strconv.FormatInt(begin.UnixMilli(), 10))
+	}
+	if !end.IsZero() {
+		params.Set("end", strconv.FormatInt(begin.UnixMilli(), 10))
+	}
+	interval := ok.GetIntervalEnum(period)
+	if interval != "" {
+		params.Set("period", interval)
+	}
+	var response [][2]string
+	err := ok.SendHTTPRequest(ctx, exchange.RestSpot, getMarginLendingRatioEPL, http.MethodGet, common.EncodeURLValues(tradingMarginLoanRatio, params), nil, &response, false)
+	if err != nil {
+		return nil, err
+	}
+	lendingRatios := []MarginLendRatioItem{}
+	for x := range response {
+		if len(response[x]) != 2 {
+			return nil, errMalformedData
+		}
+		timestamp, err := strconv.Atoi(response[x][0])
+		if err != nil {
+			return nil, err
+		}
+		ratio, err := strconv.ParseFloat(response[x][0], 64)
```

Comment:
This is parsing the timestamp again please change index to 1

Also please remove the `ratio <= 0` as this will return nil with no data.